metaflow/extension_support.py

```diff
@@ -525,111 +691,219 @@ def _attempt_load_module(module_name):
         return extension_module
 
 
-def _get_extension_config(tl_pkg, extension_point, config_module):
-    module_name = ".".join([EXT_PKG, tl_pkg, extension_point])
-    if config_module is not None:
-        _ext_debug("\t\tAttempting to load '%s'" % config_module)
-        extension_module = _attempt_load_module(config_module)
+def _get_extension_config(distribution_name, tl_pkg, extension_point, config_module):
+    if config_module is not None and not config_module.endswith("__init__"):
+        module_name = config_module
     else:
-        _ext_debug("\t\tAttempting to load '%s'" % module_name)
-        extension_module = _attempt_load_module(module_name)
+        module_name = ".".join([EXT_PKG, tl_pkg, extension_point])
+
+    _ext_debug("\t\tAttempting to load '%s'" % module_name)
+
+    extension_module = _attempt_load_module(module_name)
+
     if extension_module:
+        # We update the path to this module; this will be helpful if/when we need
+        # to package modules so we have the base path for them.
+        # Note that we cannot always get this initially because, in the case of Conda
+        # and/or remote execution, the path of the files will change (Conda does symlinks
+        # and creates intermediate directories and remote puts everything in one place)
+        if _all_packages[distribution_name]["root_path"] is None:
+            root_path = "/".join(
+                extension_module.__file__.split("/")[: -len(module_name.split("."))]
```

Comment:
Ah yes, for the cards thing that will cause issue (because that can be a ns package which I will have to handle separately). Gah. I will add a special case for that one as well. In all other cases, there should be a single file (which will either be the `__init__.py` file or the `mfextinit_X.py`.